src/main/java/ch/njol/skript/expressions/ExprItemCooldown.java

```diff
@@ -0,0 +1,140 @@
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.entity.HumanEntity;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Item Cooldown")
+@Description({"Set a cooldown on the specified material for a certain amount of ticks. ticks. 0 ticks will result in the removal of the cooldown."})
+@Examples({"on right click using stick:" +
+		"\tset item cooldown of player's tool for player to 1 minute",
+		"\tset item cooldown of stone and grass for all players to 20 seconds",
+		"\treset item cooldown of cobblestone and dirt for all players"})
+@Since("INSERT VERSION")
+public class ExprItemCooldown extends SimpleExpression<Timespan> {
+	
+	static {
+		Skript.registerExpression(ExprItemCooldown.class, Timespan.class, ExpressionType.COMBINED, 
+			"[item] cooldown of %itemtypes% for %players%",
+			"%itemtypes%[[']s] [item] cooldown for %players%");
+	}
+	
+	private Expression<HumanEntity> players;
+	private Expression<ItemType> itemtypes;
+	
+	@Override
+	@SuppressWarnings({"null", "unchecked"})
+	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
+		players = (Expression<HumanEntity>) exprs[1];
+		itemtypes = (Expression<ItemType>) exprs[0];
+		return true;
+	}
+	
+	@Override
+	@SuppressWarnings("null")
+	protected Timespan[] get(final Event e) {
+		HumanEntity[] players = this.players.getArray(e);
+		ItemType[] itemtypes = this.itemtypes.getArray(e);
+		if (players.length == 0 || itemtypes.length == 0)
+			return new Timespan[0];
+
+		int size = players.length * itemtypes.length;
+		
+		Timespan[] timespan = new Timespan[size];
+		
+		int i = 0;
+		for (HumanEntity he : players) {
+			for (ItemType it : itemtypes) {
+				timespan[i] = Timespan.fromTicks_i(he.getCooldown(it.getMaterial()));
+				i++;
+			}
+		}
+		return timespan;
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		return (mode == ChangeMode.SET || mode == ChangeMode.RESET || mode == ChangeMode.DELETE) ? CollectionUtils.array(Timespan.class) : null; 
+	}
+
+	@Override
+	public void change(Event e, Object[] delta, ChangeMode mode) {
+		HumanEntity[] players = this.players.getArray(e);
+		ItemType[] itemtypes = this.itemtypes.getArray(e);
+		if (players.length == 0 || itemtypes.length == 0)
+			return;
+		
+		switch (mode) {
+			case RESET:
+			case DELETE:
+				for (HumanEntity he : players) {
+					for (ItemType it : itemtypes) {
+						he.setCooldown(it.getMaterial(), 0);
+					}
+				}
+				break;
+			case SET:
+				assert delta[0] != null;
+				if (!(delta[0] instanceof Timespan))
+					return;
+				
+				for (HumanEntity he : players) {
+					for (ItemType it : itemtypes) {
+						he.setCooldown(it.getMaterial(), (int) ((Timespan) delta[0]).getTicks_i());
+					}
+				}
+				break;
+			default:
+				assert false;
+		}
+	}
+
+	@Override
+	public boolean isSingle() {
+		return false;
```

Comment:
```suggestion
		return players.isSingle() && itemtypes.isSingle();
```

I think this would work better since it can be single in this case